include/boost/url/grammar/detail/impl/recycled.ipp

```diff
@@ -61,30 +60,36 @@ void
 recycled_add_impl(
     std::size_t n) noexcept
 {
-    std::lock_guard<
-        std::mutex> m(all_reports_.m);
-
     auto& a = all_reports_;
 
-    a.count++;
-    if( a.count_max < a.count)
-        a.count_max = a.count;
-
-    a.bytes += n;
-    if( a.bytes_max < a.bytes)
-        a.bytes_max = a.bytes;
-
-    if( a.alloc_max < n)
-        a.alloc_max = n;
+    std::size_t new_count = ++a.count;
+    std::size_t old_count_max = a.count_max;
+    while (
+        old_count_max < new_count &&
+        !a.count_max.compare_exchange_weak(
+            old_count_max, new_count))
+    {}
+
+    std::size_t new_bytes = a.bytes.fetch_add(n);
```

Comment:
fetch_add returns the old value, not the new one. `a.bytes += n`

Reply:
:man_facepalming:

Reply:
This?

https://github.com/boostorg/url/pull/691